keanu-project/src/main/java/io/improbable/keanu/algorithms/Statistics.java

```diff
@@ -0,0 +1,39 @@
+package io.improbable.keanu.algorithms;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Statistics {
```

Comment:
Sorry this comes late but I just had a thought.

Do we always expect that the lists should all be the same length? (because at each sample step we record every statistic).

If yes:

We could enforce that, but representing Statistics as a `List` of `Map`s instead of a `Map` of `List`s. But I expect you'll tell me that you did it this way round for better performance of `get` and `average`? 

Would it be useful to have a method that returns a boolean to tell you if all the lists are the same length? Or even a `getLength` method that throws if they're not the same length?